archinstall/lib/user_interaction.py

```diff
@@ -98,22 +98,51 @@ def sig_handler(sig :int, frame :Any) -> None:
 
 	return True
 
+def is_strong_password(passwd :str) -> bool:
+
+	if not (
+		any(char.isdigit() for char in passwd)
+		and any(char.isupper() for char in passwd)
+		and any(char.islower() for char in passwd)
+		and len(passwd) >= 8
+	):
+
+		log(
+            " * Your password too weak *\n",
+            "Strong passwords must contain lowercase, uppercase,",
+            "numbers and longer than 8 letters.",
+            fg="yellow",
+        )
+		time.sleep(2) # TODO: This is a hack, we should have a better way to do this.
```

Comment:
Why is there a sleep here?

Reply:
Unfortunately, the confirmation comes on the log, so I put it.

Reply:
Can we do this instead? If a user enters a weak password, they get a prompt saying that a weak password was provided and asking if they want to continue using the same selection system we have for the menus.

Reply:
> Can we do this instead? If a user enters a weak password, they get a prompt saying that a weak password was provided and asking if they want to continue using the same selection system we have for the menus.

First I wanted to do this using 'input'. But I thought it might not be interesting. If you have a suggestion, I will gladly implement it.

Reply:
We have a TextField I believe, that supports translation. We should use that :) @svartkanin knows most about this

Reply:
You can remove the log and the sleep and move the current log text directly into the confirmation prompt that way it will be displayed on top.

Reply:
Merged :)

Reply:
So in this case we only use `select` class?